File: qualifier-week-4/graded-assignments/GrPA1 - Graded.lua
function reversedNumber(num)
    local reversed = 0

    while num ~= 0 do
        local remainder = num % 10
        reversed = reversed * 10 + remainder
        num = math.floor(num / 10)
    end
    return reversed
end

local num = tonumber(io.read())

if not num then
    print("Invalid Input: Failure")
    return
end

local original = num


if original == tonumber(reversedNumber(num)) then
    print(original, "is a palindrome number")
else
    print(original, "is not a palindrome number")
end

File: qualifier-week-4/graded-assignments/GrPA1 - Graded.c
#include <stdio.h>

int main() {
    int num, reversed = 0, remainder, original;

    scanf("%d", &num);

    original = num;

    while (num != 0) {
        remainder = num % 10;
        reversed = reversed * 10 + remainder;
        num /= 10;
    }

    if (original == reversed) {
        printf("%d is a palindrome number", original);
    } else {
        printf("%d, is not a palindrome number", original);
    }
    return 0;
}
